app/src/main/java/factorio/train/analyser/Matrix.java

```diff
@@ -33,6 +33,8 @@ public Matrix(String encodedJson){
         setMatrix();
     }
 
+    public Matrix() {}
```

Comment:
Why do you need that constructor?

Reply:
later probably for UI stuff. But isnt it common case in java to have an construktor anyway?